pkg/osm/osm_controller_http_server.go

```diff
@@ -0,0 +1,144 @@
+}
+
+// GetOSMControllerPods TODO remove once osm PR for exporting GetOSMControllerPods through osm/pkg/cli is merged.
+func GetOSMControllerPods(clientSet kubernetes.Interface, ns string) *corev1.PodList {
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"app": constants.OSMControllerName}}
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	}
+	podList, _ := clientSet.CoreV1().Pods(ns).List(context.TODO(), listOptions)
+	return podList
+}
+
+// Run implements common.Runnable
+func (check ControllerHTTPServerEndpointsCheck) Run() outcomes.Outcome {
+	// TODO replace with cli.GetOSMControllerPods(...) once osm repo PR is merged
+	controllerPods := GetOSMControllerPods(check.client, check.osmControlPlaneNamespace.String())
+
+	anyControllerPodsExist := false
+	for _, controllerPod := range controllerPods.Items {
+		anyControllerPodsExist = true
+
+		conf, err := check.actionConfig.RESTClientGetter.ToRESTConfig()
+		if err != nil {
+			return outcomes.Fail{Error: errors.Errorf("failed to get REST config from Helm %s", err)}
+		}
+		dialer, err := k8s.DialerToPod(conf, check.client, controllerPod.Name, controllerPod.Namespace)
+		if err != nil {
+			return outcomes.Fail{Error: errors.Errorf("error setting up port forwarding: %s", err)}
+		}
+		portForwarder, err := k8s.NewPortForwarder(dialer, fmt.Sprintf("%d:%d", check.localPort, constants.OSMHTTPServerPort))
+		if err != nil {
+			return outcomes.Fail{Error: errors.Errorf("error setting up port forwarding: %s", err)}
+		}
+
+		err = portForwarder.Start(func(pf *k8s.PortForwarder) error {
+			defer pf.Stop()
+			controllerHTTPServerURL := fmt.Sprintf("http://localhost:%d", check.localPort)
+
+			err = checkControllerHealthReadiness(controllerHTTPServerURL)
+			if err != nil {
+				return err
+			}
+
+			err = checkControllerHealthLiveness(controllerHTTPServerURL)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+		if err != nil {
+			return outcomes.Fail{Error: err}
+		}
+	}
+
+	if !anyControllerPodsExist {
+		return outcomes.Fail{Error: ErrorNoControllerPodsExistInNamespace}
+	}
+
+	return outcomes.Pass{}
+}
+
+// Suggestion implements common.Runnable.
+func (check ControllerHTTPServerEndpointsCheck) Suggestion() string {
+	panic("implement me")
+}
+
+// FixIt implements common.Runnable.
+func (check ControllerHTTPServerEndpointsCheck) FixIt() error {
+	panic("implement me")
+}
+
+func checkControllerHealthReadiness(controllerHTTPServerURL string) error {
+	// TODO replace "/health/ready" with constants.HTTPServer... from osm repo once PR over there is merged.
+	url := fmt.Sprintf("%s%s", controllerHTTPServerURL, "/health/ready")
+	respBody, err := utils.GetResponseBody(url)
+	if err != nil {
+		return errors.Errorf("osm-controller health readiness check failed: %s", err)
+	}
+
+	if respBody != "Service is ready" {
+		return ErrorControllerNotReady
+	}
+	return nil
```

Comment:
Checking the response code (should be `200`) is sufficient.
We certainly should not be checking the response body. Response body should mostly be for humans to understand things went ok. Machines should be checking the response codes and headers in this particular scenario.